File: PDOO/p4/civitas_rb/lib/casilla_calle.rb
module Civitas

class Casilla_calle < Casilla

  def initialize(titulo)
    super(titulo.nombre)
    @importe = titulo.preciocompra
    @titulopropiedad = titulo
    @nombre = super.nombre
  end
  
  def recibejugador(iactual, todos)
    if(jugador_correcto(iactual, todos)==true)
      
      informe(iactual, todos)
      if(!@titulopropiedad.tienepropietario)
        
        todos.at(iactual).puede_comprar_casilla
        todos.at(iactual).puedeComprar = true
      else
        
        @titulopropiedad.tramitaralquiler(todos.at(iactual))
        todos.at(iactual).puedeComprar=false
      end
    end
  end
  
  def to_string()
    return "#{super}, { importe= #{@importe} }"
  end
  
  attr_reader :titulopropiedad
    
end
end

File: PDOO/p4/Civitas/src/civitas/CasillaCalle.java

package civitas;

import java.util.ArrayList;

public class CasillaCalle extends Casilla {
    private float importe;
    
    private TituloPropiedad tituloPropiedad;
    
    CasillaCalle(TituloPropiedad titulo){
        //Para crear la casilla de tipo calle
        super(titulo.getNombre());
        importe = titulo.getPrecioCompra();
        tituloPropiedad = titulo;
    }
    
    
    @Override
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if (super.jugadorCorrecto(iactual, todos)){
            super.informe(iactual, todos);
            if (tituloPropiedad.tienePropietario() == false){
                todos.get(iactual).puedeComprarCasilla();
                todos.get(iactual).puedeComprar = true;
            } else {
                tituloPropiedad.tramitarAlquiler(todos.get(iactual));
                todos.get(iactual).puedeComprar = false;
            }
        }
    }
    
    TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
    
    @Override
    public String toString(){
        return "Casilla{ nombre=" + super.getNombre() + ", importe=" + importe + " }";
    }
    
}
